Primary rolecrud

varunasingh opened the pull request.

varunasingh committed:
Initial commit (WIP) for Role Assignment CRUD

varunasingh committed:
WIP EntityRoleEdit

varunasingh committed:
Role CRUD and Person Role assigment feature complete

varunasingh committed:
Added role db call test in presenter for Preson edit and detail

varunasingh committed:
Merge branch 'primary' into primary-rolecrud

varunasingh committed:
Tests

varunasingh committed:
code clean

varunasingh committed:
Added admin check for Roles and Permissions in Person Edit

varunasingh committed:
Added admin check for Roles and permissions

varunasingh committed:
Merge branch 'primary' into primary-rolecrud

varunasingh committed:
Presenter tests fix

varunasingh committed:
Merge branch 'primary' into primary-rolecrud
Moved from 38 to 39

mikedawson commented on lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/UmAppDatabase.kt:
```diff
@@ -1259,12 +1264,83 @@ abstract class UmAppDatabase : DoorDatabase(), SyncableDoorDatabase {
+
+                } else if (database.dbType() == DoorDbType.POSTGRES){
+                    //Person
+                    database.execSQL("ALTER TABLE Person RENAME to Person_OLD")
```
I would recommend using Postgres' improved support for table modification here instead.

mikedawson replied:
Also: for any table (e.g. PersonGroup) where on Sqlite you had to rename/drop tables, you need to add the SQLite triggers back (otherwise it won't sync again)

e.g.
```
_stmt.executeUpdate("""
          |CREATE TRIGGER UPD_43
          |AFTER UPDATE ON PersonGroup FOR EACH ROW WHEN
          |(SELECT CASE WHEN (SELECT master FROM SyncNode) THEN 
          |(NEW.groupMasterCsn = 0 
          |OR OLD.groupMasterCsn = NEW.groupMasterCsn
          |)
          |ELSE
          |(NEW.groupLocalCsn = 0  
          |OR OLD.groupLocalCsn = NEW.groupLocalCsn
          |) END)
          |BEGIN 
          |UPDATE PersonGroup SET groupLocalCsn = 
          |(SELECT CASE WHEN (SELECT master FROM SyncNode) THEN NEW.groupLocalCsn 
          |ELSE (SELECT MAX(MAX(groupLocalCsn), OLD.groupLocalCsn) + 1 FROM PersonGroup) END),
          |groupMasterCsn = 
          |(SELECT CASE WHEN (SELECT master FROM SyncNode) THEN 
          |(SELECT MAX(MAX(groupMasterCsn), OLD.groupMasterCsn) + 1 FROM PersonGroup)
          |ELSE NEW.groupMasterCsn END)
          |WHERE groupUid = NEW.groupUid
          |; END
          """.trimMargin())
          _stmt.executeUpdate("""
          |CREATE TRIGGER INS_43
          |AFTER INSERT ON PersonGroup FOR EACH ROW WHEN
          |(SELECT CASE WHEN (SELECT master FROM SyncNode) THEN 
          |(NEW.groupMasterCsn = 0 
          |
          |)
          |ELSE
          |(NEW.groupLocalCsn = 0  
          |
          |) END)
          |BEGIN 
          |UPDATE PersonGroup SET groupLocalCsn = 
          |(SELECT CASE WHEN (SELECT master FROM SyncNode) THEN NEW.groupLocalCsn 
          |ELSE (SELECT MAX(groupLocalCsn) + 1 FROM PersonGroup) END),
          |groupMasterCsn = 
          |(SELECT CASE WHEN (SELECT master FROM SyncNode) THEN 
          |(SELECT MAX(groupMasterCsn) + 1 FROM PersonGroup)
          |ELSE NEW.groupMasterCsn END)
          |WHERE groupUid = NEW.groupUid
          |; END
          """.trimMargin())
```

varunasingh committed:
Added sort to Role list
Fixed issue on admin check for role assignment
Fixed roles and assignment text

varunasingh committed:
Updated to create triggers in database migration
Updated postgres add column for person update
Removed clickable background on roles and permissions's icon
Added search functionality on Role List
Updated role entity edit's error message to be generic.

varunasingh committed:
Fixed role picking from role list.
Fixed erUid check
added check for active on Roles and Assignments

varunasingh committed:
Merge branch 'primary' into primary-rolecrud

varunasingh committed:
Merge branch 'primary' into primary-rolecrud

varunasingh committed:
Fix marking loading

mikedawson committed:
Merge branch 'primary' into primary-rolecrud

varunasingh committed:
Merge branch 'primary' into primary-rolecrud

varunasingh committed:
Fixed crash on PersonEdit

varunasingh committed:
Merge branch 'primary-rolecrud' of github.com:UstadMobile/UstadMobile into primary-rolecrud
